manager/controllers/app/m4dapplication_controller.go

```diff
@@ -364,6 +366,11 @@ func (r *M4DApplicationReconciler) reconcile(applicationContext *app.M4DApplicat
 		return ctrl.Result{RequeueAfter: 2 * time.Second}, allocErr
 	}
 	// generate blueprint specifications (per cluster)
+	if len(instances) == 0 {
```

Comment:
What it it wasn't empty before? Shouldn't you clear up existing blueprints?

Reply:
Added clean-up